shell/browser/api/electron_api_session.cc

```diff
@@ -591,9 +591,15 @@ void Session::AllowNTLMCredentialsForDomains(const std::string& domains) {
 void Session::SetUserAgent(const std::string& user_agent,
                            gin::Arguments* args) {
   browser_context_->SetUserAgent(user_agent);
-  content::BrowserContext::GetDefaultStoragePartition(browser_context_.get())
-      ->GetNetworkContext()
-      ->SetUserAgent(user_agent);
+  auto* network_context = content::BrowserContext::GetDefaultStoragePartition(
+                              browser_context_.get())
+                              ->GetNetworkContext();
+  network_context->SetUserAgent(user_agent);
+
+  std::string accept_lang;
+  if (args->GetNext(&accept_lang)) {
+    network_context->SetAcceptLanguage(accept_lang);
```

Comment:
Should be

`network_context->SetAcceptLanguage(net::HttpUtil::GenerateAcceptLanguageHeader(accept_lang));`

Reply:
Also should add a test that gives an input of accept languages like `en-US,fr,de` , so the generated header is validated and turns out as `en-US,fr;q=0.9,de;q=0.8`